can/io/blf.py

```diff
@@ -395,6 +395,11 @@ def __init__(
             speed and compression (currently equivalent to level 6).
         """
         mode = "rb+" if append else "wb"
+        if type(file) == gzip.GzipFile:
+            raise ValueError(
+                f"The BLFWriter is currently incompatible with "
+                f"{gzip.GzipFile.__name__}s. Try using a different zip format."
+            )
```

Comment:
I'm unaware of any zip format that could be made compatible with the current implementation of `BLFWriter`.